invenio_app_ils/ui/src/invenio_app_ils/pages/backoffice/Series/SeriesForms/SeriesEditor/components/SeriesForm/SeriesForm.js

```diff
@@ -79,65 +59,61 @@ export class SeriesForm extends Component {
 
   render() {
     return (
-      <>
-        <Grid>
-          <Grid.Row centered>
-            <Header>{this.title}</Header>
-          </Grid.Row>
-        </Grid>
-
-        <Formik
-          initialValues={
-            this.formInitialData
-              ? this.prepareData(this.formInitialData.metadata)
-              : {}
-          }
-          onSubmit={this.onSubmit}
-          render={({ isSubmitting, handleSubmit }) => (
-            <Form onSubmit={handleSubmit} loading={isSubmitting}>
-              <Segment>
-                <StringField label="Title" fieldPath="title" required />
-                <SelectField
-                  label="Mode of issuance"
-                  fieldPath="mode_of_issuance"
-                  options={[
-                    {
-                      text: 'MULTIPART_MONOGRAPH',
-                      value: 'MULTIPART_MONOGRAPH',
-                    },
-                    {
-                      text: 'SERIAL',
-                      value: 'SERIAL',
-                    },
-                  ]}
-                  required
-                ></SelectField>
-                <TextField
-                  label="Abstract"
-                  fieldPath="abstract"
-                  uiProps={{ rows: 10 }}
-                />
-                <ArrayStringField
-                  label="Authors"
-                  fieldPath="authors"
-                ></ArrayStringField>
-                <SelectField
-                  multiple
-                  label="Languages"
-                  fieldPath="languages"
-                  options={this.getLanguageCodes()}
-                  uiProps={{ upward: false }}
-                ></SelectField>
-                <StringField label="Edition" fieldPath="edition" />
-                <StringField label="ISSN" fieldPath="issn" />
-              </Segment>
-              <Button color="green" disabled={isSubmitting}>
-                Submit
-              </Button>
-            </Form>
-          )}
+      <BaseForm
+        initialValues={
+          this.formInitialData
+            ? this.prepareData(this.formInitialData.metadata)
+            : {}
+        }
+        editApiMethod={this.updateSeries}
+        createApiMethod={this.createSeries}
+        successCallback={this.successCallback}
+        successSubmitMessage={this.successSubmitMessage}
+        title={this.title}
+        pid={this.pid ? this.pid : undefined}
+      >
+        <StringField label="Title" fieldPath="title" required />
+        <SelectField
+          label="Mode of issuance"
+          fieldPath="mode_of_issuance"
+          options={[
```

Comment:
Can we put this mapping in the config? It would be nice to re-use these human readable labels also in the Details and Search pages.

Reply:
WIill be addressed in the next pr for eitems along with the delivery method config refactor.